src/simulator/diff_drive_model.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * The MIT License (MIT)
+ * Copyright (c) 2011 William Woodall <wjwwood@gmail.com>
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a 
+ * copy of oftware and associated documentation files (the "Software"), 
+ * to deal in the Software without restriction, including without limitation 
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+ * and/or sell copies of the Software, and to permit persons to whom the 
+ * Software is furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and ermission notice shall be included 
+ * in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include <iostream>
+#include <sstream>
+#include <cmath>
+
+#include <tf/transform_broadcaster.h>
+#include <geometry_msgs/Twist.h>
+#include <nav_msgs/Odometry.h>
+#include "diff_drive_model.h"
+#include "shared/util/timer.h"
+
+using std::vector;
+using std::string;
+
+namespace diffdrive {
+
+CONFIG_BOOL(invert_x, "invert_linear_vel_cmds");
+CONFIG_BOOL(invert_z, "invert_angular_vel_cmds");
+CONFIG_BOOL(broadcast_tf, "broadcast_tf");
+CONFIG_FLOAT(linear_pos_accel_limit, "linear_pos_accel_limit");
+CONFIG_FLOAT(linear_neg_accel_limit, "linear_neg_accel_limit");
+CONFIG_FLOAT(angular_pos_accel_limit, "angular_pos_accel_limit");
+CONFIG_FLOAT(angular_neg_accel_limit, "angular_neg_accel_limit");
+CONFIG_FLOAT(max_angular_vel, "max_angular");
+CONFIG_FLOAT(max_linear_vel, "max_linear_vel");
+
+CONFIG_STRING(drive_topic, "drive_callback_topic");
+CONFIG_STRING(odom_topic, "diff_drive_odom_topic");
+CONFIG_FLOAT(linear_odom_scale, "linear_odom_scale");
+CONFIG_FLOAT(angular_odom_scale, "angular_odom_scale");
+
+
+DiffDriveModel::DiffDriveModel(const vector<string>& config_files, ros::NodeHandle* n) :
+	RobotModel(),
+	last_cmd_(),
+	t_last_cmd_(0),
+	angular_error_(0, 1),
+	config_reader_(config_files){
+        
+	drive_subscriber_ = n->subscribe(
+      CONFIG_drive_topic,
+      1,
+      &DiffDriveModel::DriveCallback,
+      this);
+	odom_publisher_ = n->advertise<nav_msgs::Odometry>(CONFIG_odom_topic, 1);
+    odometry_x = 0.0;
```

Comment:
Values should be initialized in the constructor using the same syntax as `t_last_cmd_`, etc.